src/citrine/informatics/predictor_evaluation_result.py

```diff
@@ -0,0 +1,245 @@
+
+    def __init__(self, value: List[PredictedVsActual]):
+        self.value = value
+
+
+class ResponseMetrics(Serializable["ResponseMetrics"]):
+    """[ALPHA] Set of metrics computed by a Predictor Evaluator for a single response.
+    Results computed for a metric can be accessed by the metric's ``__repr__`` or by the metric itself.
+
+    Parameters
+    ----------
+    metrics: Mapping[str, MetricValue]
+        Metrics computed for a single response, keyed by the metric's ``__repr__``.
+
+    """
+    metrics = properties.Mapping(properties.String, properties.Object(MetricValue), "metrics")
+    typ = properties.String('type', default='ResponseMetrics', deserializable=False)
+
+    def __init__(self, *, metrics: Mapping[str, MetricValue]):
+        self.metrics = metrics
+
+    def __getitem__(self, item):
+        if isinstance(item, str):
+            return self.metrics[item]
+        elif isinstance(item, PredictorEvaluationMetric):
+            return self.metrics[repr(item)]
+        else:
+            raise TypeError("Cannot index ResponseMetrics with a {}".format(type(item)))
+
+
+class PredictorEvaluationResult(PolymorphicSerializable["PredictorEvaluationResult"]):
+    """[ALPHA] A Citrine Predictor Evaluation Result represents a set of metrics
+    computed by a Predictor Evaluator.
+
+    """
+
+    @classmethod
+    def get_type(cls, data) -> Type[Serializable]:
+        """Return the subtype."""
+        return {
+            "CrossValidationResult": CrossValidationResult,
+        }[data["type"]]
+
+    def evaluator(self) -> PredictorEvaluator:
+        """Evaluator that produced the result."""
+        raise NotImplementedError()
+
+    def responses(self) -> Set[str]:
+        """Predictor responses that were evaluated."""
+        raise NotImplementedError()
+
+    def metrics(self) -> Set[PredictorEvaluationMetric]:
+        """Metrics computed for predictor responses."""
+        raise NotImplementedError()
+
+
+class CrossValidationResult(Serializable["CrossValidationResult"], PredictorEvaluationResult):
+    """[ALPHA] Result of performing a cross-validation evaluation on a predictor.
+
+    Parameters
+    ----------
+    evaluator: PredictorEvaluator
+        Evaluator that produced this result.
+    response_results: Mapping[str, ResponseMetrics]
+        Map from response key to all metrics compute for that response.
+
+    """
+    _evaluator = properties.Object(PredictorEvaluator, "evaluator")
+    _response_results = properties.Mapping(properties.String, properties.Object(ResponseMetrics), "response_results")
+    typ = properties.String('type', default='CrossValidationResult', deserializable=False)
+
+    def __init__(self, *,
+                 evaluator: PredictorEvaluator,
+                 response_results: Mapping[str, ResponseMetrics]):
+        self._evaluator = evaluator
+        self._response_results = response_results
+
+    def __getitem__(self, item):
```

Comment:
This behavior should probably be documented in the class docstring.

Reply:
👍